### ISSUE TITLE
Add Split Softmax

### ISSUE THREAD

**Issue by AMindToThink** [pytorch/pytorch#153035]
### 🚀 The feature, motivation and pitch

Transformer models often forget their system prompts when processing long text due to the long distances between the source of the information and its destination.

The Split Softmax function is a modification of softmax for use in attention that promotes the model to keep paying attention to the system prompt. It was developed to combat instruction drift and attention decay. 

It is described in this paper: [Measuring and Controlling Instruction (In)Stability
in Language Model Dialogs](https://arxiv.org/pdf/2402.10962)

I am doing research involving this function, and I would like to implement split softmax as an activation function in Pytorch, if it is deemed worthwhile.

**Comment by janeyx99** [comment]
Thanks for the suggestion PyTorch Core has some pretty strict rules when it comes to adding a new Module or Activation Function: https://github.com/pytorch/pytorch/wiki/Developer-FAQ#i-have-a-new-function-or-feature-id-like-to-add-to-pytorch-should-it-be-in-pytorch-core-or-a-library-like-torchvision

In particular, the way to go here may be to put split softmax in a separate library. Please don't hesitate to let us know if there is any low level feature that you're missing to implement this!
